clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/NamespaceProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.model;
+
+import com.netflix.spinnaker.clouddriver.documentation.Empty;
+import java.util.Map;
+
+public interface NamespaceProvider {
+
+  /**
+   * Looks up all of the namespaces for a particular account. Keyed on account name.
+   *
+   * @param account name
+   * @return map of namespace-names to namespaces objects.
+   */
+  @Empty
+  Map<String, Object> getNamespaces(String account);
```

Comment:
`Map<String, Object>` is not a helpful return type - we shouldn't be propagating this type of method signature to new places in the codebase wherever possible. Is there a common interface we can implement here?

Reply:
I agree, and yes, I hope we could think about a common interface.

If we created a Namespace interface, to begin it could have a single method signature ->
 `String getName()`
 
This would satisfy the most basic interactions and leave room for further development.

Reply:
Sounds good.